python/lib/metrics/dmod/metrics/communication.py

```diff
@@ -7,38 +9,183 @@
 import logging
 import traceback
 
+from datetime import datetime
+from pprint import pprint
 from collections import abc as abstract_collections
 
 MESSAGE = typing.Union[bytes, str, typing.Dict[str, typing.Any], typing.Sequence, bool, int, float]
-MESSAGE_HANDLER = typing.Callable[[MESSAGE], typing.NoReturn]
-REASON_TO_WRITE = typing.Union[str, typing.Dict[str, typing.Any]]
+MessageHandler = typing.Callable[[MESSAGE], typing.NoReturn]
+ReasonToWrite = typing.Union[str, typing.Dict[str, typing.Any]]
 
 
-class Verbosity(enum.IntEnum):
+class Verbosity(enum.Enum):
     """
     An enumeration detailing the density of information that may be transmitted, not to logs,
     but through things like streams and communicators
     """
-    QUIET = enum.auto()
+    QUIET = "QUIET"
     """Emit very little information"""
 
-    NORMAL = enum.auto()
+    NORMAL = "NORMAL"
     """Emit a baseline amount of information"""
 
-    LOUD = enum.auto()
+    LOUD = "LOUD"
     """Emit a lot of detailed (often diagnostic) information"""
 
-    ALL = enum.auto()
+    ALL = "ALL"
     """Emit everything, including raw data"""
 
+    @classmethod
+    def get_by_name(cls, name: str) -> Verbosity:
```

Comment:
That only works here by chance - change the type to just about anything else and now it doesn't work. This function is supposed to search based on name, not value.

I'm not prepared to push back _too_ hard on this one since the only stopper is a break in mental model/philosophy, so if you feel really strongly I'll change it.